plugins/Capacity.py

```diff
@@ -0,0 +1,131 @@
+
+###########################################################################
+##                                                                       ##
+## Copyrights Éric Gillet 2020                                           ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+
+class Capacity(Plugin):
+    def init(self, logger):
+        Plugin.init(self, logger)
+
+        self.errors[35001] = self.def_class(
+            item=3500,
+            level=2,
+            tags=["tag"],
+            title=T_("Invalid capacity value"),
+            detail=T_("""A capacity tag value is incorrect."""),
+            resource="https://wiki.openstreetmap.org/wiki/Key:capacity",
+        )
+
+    def node(self, data, tags):
+        if ("capacity" not in tags
+                # Ignore errors that should be reported by generic prefix analysers
+                or tags["capacity"] == ""):
+            return
+        try:
+            total_capacity = int(tags["capacity"])
+        except ValueError:
+            return {
+                "class": 35001,
+                "subclass": 0,
+                "text": T_('Invalid "{0}" value: {1}', "capacity", tags["capacity"]),
+            }
+
+        if total_capacity < 0:
+            return {
+                "class": 35001,
+                "subclass": 0,
+                "text": T_('Invalid "{0}" value: {1}', "capacity", total_capacity),
+            }
+
+        for key, value in tags.items():
+            if (
+                not key.startswith("capacity:")
+                # Ignore errors that should be reported by generic prefix analysers
+                or key == "capacity:"
+                or value == ""
+                or value in ("yes", "no")
+            ):
+                continue
+
+            try:
+                capacity_int = int(value)
+            except ValueError:
+                return {
```

Comment:
You may return an array, you do not have to return on first issue.

Reply:
Thank you for the suggestion, I made it so in 3ddfae591179175b88d8e3156417625157bafd76.